solarforecastarbiter/io/reference_observations/tests/test_ref_eia.py

```diff
@@ -0,0 +1,171 @@
+    requests_mock.register_uri('POST', matcher,
+                               text=single_observation.observation_id)
+    requests_mock.register_uri('GET', matcher, content=single_observation_text)
+    status = mocker.patch(
+        'solarforecastarbiter.io.api.APISession.create_observation')
+    eia.initialize_site_observations(session, site)
+    assert status.called
+
+
+@pytest.fixture
+def mock_obs_creation(mocker):
+    mocked = mocker.patch(
+        'solarforecastarbiter.io.reference_observations.common.'
+        'create_observation')
+    return mocked
+
+
+@pytest.fixture
+def mock_obs_creation_error(mocker, mock_obs_creation):
+    mock_obs_creation.side_effect = HTTPError(
+        response=mocker.Mock(response=mocker.Mock(text="eror")))
+
+
+def test_initialize_site_obs(mock_api, mock_obs_creation, site):
+    eia.initialize_site_observations(mock_api, site)
+    mock_obs_creation.assert_called()
+
+
+def test_initialize_site_obs_http_error(
+        log, mock_api, mock_obs_creation_error, site):
+    eia.initialize_site_observations(mock_api, site)
+    assert log.error.call_count == 1
+    assert log.debug.call_count == 1
+
+
+def test_fetch(mocker, session, site):
+    status = mocker.patch(
+        'solarforecastarbiter.io.fetch.eia.get_eia_data'
+    )
+    start = pd.Timestamp('2020-01-01T0000Z')
+    end = pd.Timestamp('2020-01-02T0000Z')
+    api_key = 'bananasmoothie'
+
+    index = pd.date_range(start, end, freq="1h")
+    data = {"net_load": range(len(index))}
+    df = pd.DataFrame(index=index, data=data)
+    status.return_value = df
+
+    out = eia.fetch(session, site, start, end, eia_api_key=api_key)
+    assert status.called
+    assert not out.empty
+
+
+def test_fetch_empty(log, mocker, session, site):
+    status = mocker.patch(
+        'solarforecastarbiter.io.fetch.eia.get_eia_data'
+    )
+    start = pd.Timestamp('2020-01-01T0000Z')
+    end = pd.Timestamp('2020-01-02T0000Z')
+    api_key = 'bananasmoothie'
+    status.return_value = pd.DataFrame()
+    out = eia.fetch(session, site, start, end, eia_api_key=api_key)
+    assert status.called
+    assert log.warning.call_count == 1
+    assert out.empty
+
+
+def test_fetch_fail(mocker, session, site_no_extra):
+    start = pd.Timestamp('2020-01-01T0000Z')
+    end = pd.Timestamp('2020-01-02T0000Z')
+    api_key = 'bananasmoothie'
+    out = eia.fetch(session, site_no_extra, start, end, eia_api_key=api_key)
+    assert out.empty
+
+
+def test_initialize_site_forecasts(
+        requests_mock, mocker, session, site, single_forecast,
+        single_forecast_text, mock_list_sites):
+    matcher = re.compile(f'{session.base_url}/forecasts/.*')
+    requests_mock.register_uri('POST', matcher,
+                               text=single_forecast.forecast_id)
+    requests_mock.register_uri('GET', matcher, content=single_forecast_text)
+    status = mocker.patch(
+        'solarforecastarbiter.io.reference_observations.common.'
+        'create_forecasts')
```

Comment:
So you're suggesting something like changing to `mocker.patch('solarforecastarbiter.io.api.APISession.create_forecast')` and then revising `io.api.APISession.create_forecast()` to skip the NWP forecasts if the variable is net load?

Reply:
yes to the patch change, and create_nwp_forecasts should already skip nwp for net load, but there is a but chokes when there are no other variables. So something like returning if ``vars_`` is empty on line 533 of comman.py should resolve the error that crops up once you switch to patching ``APISession.create_forecast`` instead of ``reference_observations.common.create_forecasts``